raw_socket_listener/tcp_message.go

```diff
@@ -92,9 +92,14 @@ func (t *TCPMessage) IsMultipart() bool {
 	}
 
 	payload := t.packets[0].Data
-	m := payload[:4]
```

Comment:
Nope. Some http messages consist of multiple tcp packets. If isMultipart
returns false, it means message can be dispatched immediately (for example
simple GET request). Otherwise it will wait for rest of packets for 2
seconds.

In our case if we received packet with empty space in data as first one, it
means that packet was captured in wrong order, message is multipart, and we
should wait for next packet.

On Wednesday, 30 September 2015, Doychin <notifications@github.com> wrote:

> In raw_socket_listener/tcp_message.go
> <https://github.com/buger/gor/pull/216#discussion_r40775325>:
>
> > @@ -92,9 +92,14 @@ func (t *TCPMessage) IsMultipart() bool {
> >  	}
> >
> >  	payload := t.packets[0].Data
> > -	m := payload[:4]
>
> But this way you would miss this packet and possibly break the TCP flow.
> Am I correct?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/buger/gor/pull/216/files#r40775325>.
>


-- 
Sincerely yours Leonid Bugaev
leonsbox@gmail.com
http:// <http://buger.github.com>leonsbox.com